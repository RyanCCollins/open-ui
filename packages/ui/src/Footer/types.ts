import * as React from 'react';
import Footer from './';

export interface FooterProps extends React.Props<typeof Footer> {
  color?: string;
}
