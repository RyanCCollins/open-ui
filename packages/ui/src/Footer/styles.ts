const styled = require('styled-components').default;

export const FooterComponent = styled.footer`
  background: ${(props) => props.color};
  height: 300px;
  padding-top: 50px;
  text-align: center;
  width: 100%;
`;
