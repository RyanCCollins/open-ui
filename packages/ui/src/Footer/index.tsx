import * as React from 'react';
import { FooterComponent } from './styles';
import { FooterProps } from './types';

export default function Footer({
  color,
  children,
}: FooterProps): JSX.Element {
  const colorProp = color || '#0a0a0a';
  return (
    <FooterComponent color={colorProp}>
      {children}
    </FooterComponent>
  );
};
