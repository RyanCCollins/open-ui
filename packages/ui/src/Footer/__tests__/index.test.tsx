import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Footer from '../';
import props from './__mocks__/footerMocks.mock';

describe('<Footer />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Footer>
        <div />
      </Footer>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with different props', () => {
    const wrapper = shallow(
      <Footer {...props}>
        <div />
      </Footer>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
