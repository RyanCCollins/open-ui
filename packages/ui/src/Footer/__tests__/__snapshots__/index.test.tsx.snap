// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render with default props 1`] = `
<styled.footer
  color="#0a0a0a"
>
  <div />
</styled.footer>
`;

exports[`<Footer /> should render with different props 1`] = `
<styled.footer
  color="#0b0b0b"
>
  <div />
</styled.footer>
`;
